lib/Sema/TypeCheckNameLookup.cpp

```diff
@@ -363,7 +400,55 @@ LookupTypeResult TypeChecker::lookupMemberType(DeclContext *dc,
   if (options.contains(NameLookupFlags::IgnoreAccessibility))
     subOptions |= NL_IgnoreAccessibility;
 
-  if (!dc->lookupQualified(type, name, subOptions, this, decls))
+  auto lookupSuccess = dc->lookupQualified(type, name, subOptions, this, decls);
+
+  // If the requested type is a CodingKeys type on something which may require
+  // Encodable/Decodable synthesis, it may not have been synthesized yet. If
+  // this is the case, force synthesis and attempt lookup again.
+  if (!lookupSuccess && name == Context.Id_CodingKeys) {
+    if (auto *targetDecl = type->getAnyNominal()) {
+      auto *encodableProto = Context.getProtocol(KnownProtocolKind::Encodable);
+      auto *decodableProto = Context.getProtocol(KnownProtocolKind::Decodable);
+
+      // If the type either conforms to Encodable/Decodable directly, or
+      // inherits from something which does, the conformance check will force
+      // synthesis, which will allow us to try again.
+      NormalProtocolConformance *conformance = nullptr;
+      Optional<ProtocolConformanceRef> conformanceRef;
+      if ((conformanceRef = conformsToProtocol(type, decodableProto, targetDecl,
+                                               ConformanceCheckFlags::Used,
+                                               SourceLoc()))) {
```

Comment:
You should use the same conformance check options shown [below](https://github.com/apple/swift/pull/10723/files#diff-554fe5ac9125c2a67ed8607539330b36R521).